test/testdata/bound_proc.rb

```diff
@@ -0,0 +1,63 @@
+# typed: true
+
+module Concern
+  def included(&block)
+    @block = block
+  end
+end
+
+module Readable
+  extend Concern
+
+  included do
+    before_create :do_this
+  # ^^^^^^^^^^^^^^^^^^^^^^ error: Method `before_create` does not exist on `T.class_of(Readable)`
+  end
```

Comment:
These tests don't look like the tests that you had written in your PR description.

We very much prefer test cases that mimic real-world usage.

Doesn't Sorbet already have a `rewriter` pass to support `Concern`'s `included do` pattern? I get that it's possible to use `T.bind` for this example, but I don't think that in practice people will be using `T.bind` for this particular use case much, over the rewriter pass (correct me if I'm mistaken).

With that in mind, the example in your PR description with the `before_create` hook looked more representative, so I'd love if you could update your test with those examples.

Reply:
I updated the tests a little bit, however, I must argue that they do reflect real world usage. The `Concern` rewriter is for handling class methods - not the included block which Sorbet still cannot understand the correct binding.

Additionally, defining callbacks inside included blocks, such as `before_create` is a common use case of Rails. I added your tests and also changed the examples a bit, but again they do reflect day to day usage of Rails.

Reply:
Cool thanks for the context!